### ISSUE TITLE
Creating a new locale doesn't pre-fill the non-internationalized fields

### ISSUE THREAD

**Issue by mauriciabad** [strapi/strapi#17860]
### Required System information

- Node.js version: v18.17.1
- NPM version: _idk, almost the latest one_
- Strapi version:  4.12.7
- Database: mysql
- Operating system: linux (ubuntu)
- Is your project Javascript or Typescript: Typescript

### Describe the bug

Creating a new locale doesn't pre-fill the non-internationalized fields. But when you refresh the page, they get filled. This is a confusing behavior.

#### Setup strapi instance
1. Create a new strapi project with quickstart
1. Install internationalization plugin.
1. Add a second locale in settings.
1. Create a collection-type with the content-type builder with Internationalization enabled.

#### Reproduce the error
1. Click `Create a new entry` on the collection-type just created.
1. Fill the fields and click save (and publish if was enabled)
1. Click the `locale` dropdown and select another locale what doesn't have an entry yet.
1. See error. All fields are empty, including the non-internationalized fields.
   - [ALTERNATIVE PATH]
   1. See another error. Notice that if you open the `locale` dropdown again, the previous locale is marked as if it didn't have an entry, although it does. 
   1. If here you click the previous locale, it takes you to an empty new page with the locale that should be filled but it's not. 
   1. If you reload, fields are also filled now. But the page is still `Create an entry` instead of the normal editing view.
1. Reload the page
1. The non-internationalized fields are magically filled now. And the `locale` dropdown displays the correct state for the translations.

### Expected behavior

The content of the page before and after reloading should be exactly the same. 

Meaning that:
- Non-internationalized fields are pre-filled when creating a new entry.
- The `locales` dropdown should display the correct state for the translations at all time.

### Screenshots

I made a video :)

[screen-capture.webm](https://github.com/strapi/strapi/assets/12821361/a1d4d66b-ad0a-425c-9018-adcc8f207d3d)

### Code snippets

If applicable, add code samples to help explain your problem.

### Additional context

These are the dependencies I have installed:

```json
"dependencies": {
  "@creazy231/strapi-plugin-apollo-sandbox": "0.0.13",
  "@strapi/plugin-graphql": "4.12.7",
  "@strapi/plugin-i18n": "4.12.7",
  "@strapi/plugin-seo": "1.9.2",
  "@strapi/plugin-users-permissions": "4.12.7",
  "@strapi/provider-upload-local": "4.12.7",
  "@strapi/strapi": "4.12.7",
  "mysql": "2.18.1",
  "strapi-plugin-config-sync": "1.1.3",
  "strapi-plugin-multi-select": "1.2.2"
},
```

**Comment by bytejan** [comment]
I have encountered the same issue subsequent to updating my npm packages. My dependencies are as follows:

```
"dependencies": {
    "@strapi/plugin-color-picker": "^4.12.7",
    "@strapi/plugin-documentation": "^4.12.7",
    "@strapi/plugin-i18n": "4.12.7",
    "@strapi/plugin-seo": "^1.9.2",
    "@strapi/plugin-users-permissions": "4.12.7",
    "@strapi/strapi": "^4.12.7",
    "better-sqlite3": "8.0.1",
    "strapi-plugin-duplicate-button": "^1.1.13",
    "strapi-plugin-multi-select": "^1.2.2",
    "strapi-plugin-navigation": "^2.2.11"
}
```

node: v16.17.0
package manager: Tried with yarn and npm

Even when I attempted to initiate a new project using npx create-strapi-app@latest, I encountered the same issue. The problem persists in this fresh project as well (both yarn / npm) 


The option to copy the language appears when you create a new language for the collection-type

**Comment by akhmadshin** [comment]
It was working in version 4.9.1 (and maybe above)

**Comment by derrickmehaffy** [comment]
If a field is non-localized it should be automatically synced to all locales. Regardless if it was working this is the intended way the feature should work.

**Comment by akhmadshin** [comment]
@derrickmehaffy 
Problem is that middleware "ContentManager/CrudReducer/INIT_FORM" fires before middleware "ContentManager/EditViewLayoutManager/SET_LAYOUT", which leads to error in file addCommonFieldsToInitialDataMiddleware.js (line 50)
{ model: currentLayout.contentType.uid, id: relatedEntityId, locale }
can't read property contentType of undefined

To fix it, I suggest to wrap "init" function in setTimeOut 0. File CollectionTypeFormWrapper/index.js (147 line)
```
const init = async () => {
      setTimeout(() => {
        dispatch(getData());
        dispatch(initForm(rawQuery));
      }, 0);
    };
```

**Comment by SalahAdDin** [comment]
> @derrickmehaffy Problem is that middleware "ContentManager/CrudReducer/INIT_FORM" fires before middleware "ContentManager/EditViewLayoutManager/SET_LAYOUT", which leads to error in file addCommonFieldsToInitialDataMiddleware.js (line 50) { model: currentLayout.contentType.uid, id: relatedEntityId, locale } can't read property contentType of undefined
> 
> To fix it, I suggest to wrap "init" function in setTimeOut 0. File CollectionTypeFormWrapper/index.js (147 line)
> 
> ```
> const init = async () => {
>       setTimeout(() => {
>         dispatch(getData());
>         dispatch(initForm(rawQuery));
>       }, 0);
>     };
> ```
> 
> It fixes both bugs from your video

Ought to an MR, cool!

**Comment by mauriciabad** [comment]
Wrapping the function inside a `setTimeout` works to prove the hypothesis, but seems sketchy as a final solution. _(At least I would add a comment explaining why there's a `setTimeout` set to 0)_

Consider that if this 2 middlewares are not firing in the right order, maybe others are having the same issue. So maybe it's worth checking why they are fired in an unexpected order, and change their trigger.

**Comment by derrickmehaffy** [comment]
Indeed a set timeout is not a solution, it's a bandaid at best.

**Comment by derrickmehaffy** [comment]
The middlewares have a specific order and either they are out of order or someone set something as async when it shouldn't be.

**Comment by akhmadshin** [comment]
That bug happens only on soft navigation (when you select locale in dropdown).  If I select locale and reload page, middlewares will fire in correct order

**Comment by AndreaCeccatoUlulab** [comment]
This is a big problem, especially when you have mandatory fields and a data entry to perform with many fields.
Reloading the page is an inconvenient solution, only temporary. I hope it gets resolved as soon as possible.
It is important that all types of fields are pre-filled, especially the mandatory ones.
Thanks to the Strapi community.

**Comment by SimonStorlSchulke** [comment]
I can 

> > It was working in version 4.9.1 (and maybe above)
> 
> I don't recall this ever working, even since version 3.

This was working previously. Here is a video of our Strapi instance running in 4.10.5.

https://github.com/strapi/strapi/assets/25198913/297d7894-1130-4c98-a11c-96ecb12e5cb3

After the update to 4.13.x, it broke.

**Comment by Eventyret** [comment]
Escalated due to confirmation from TID 4804

**Comment by SalahAdDin** [comment]
> Escalated due to confirmation from TID 4804

finally!!!

**Comment by arvidinge** [comment]
Still an issue, im on 5.6.0

I have 2 locales for a content-type, 3 localized fields and one non-localized.
Create an entry in one of the locales with all fields filled out.
Click the dropdown in top right corner and click `Create English(en) locale`
The non-localized field is not prefilled.
Hard reload does not prefill the field.

**Comment by vbartalis** [comment]
> Still an issue, im on 5.6.0

I found another issue related to the inconsistent locale behavior in v5. To me it seems to be an entirely different bug, based on its behavior.  I'll put it in here for reference #22080

**Comment by SalahAdDin** [comment]
> Still an issue, im on 5.6.0
> 
> I have 2 locales (swedish, english) for a content-type, 3 localized fields and one non-localized. Create an entry in swedish with all fields filled out. Click the dropdown in top right corner and click `Create English(en) locale` The non-localized field is not prefilled. Hard reload does not prefill the field.

So we need to create a new issue about it.

**Comment by sq-tupai** [comment]
Still a issue on 5.30.0, and its not only the regular fields but relationships.

**Comment by vbartalis** [comment]
Seems like this issue is a duplicate of https://github.com/strapi/strapi/issues/21850 and it was fixed in this PR: https://github.com/strapi/strapi/pull/24659.
It should be fixed in v5.30.1, which will come out sometime today.
